monolith/views/users.py

```diff
@@ -11,17 +11,20 @@ def _users():
     return render_template("users.html", users=users)
 
 
-@users.route('/create_user', methods=['GET', 'POST'])
+@users.route('/signup', methods=['GET', 'POST'])
 def create_user():
+    if current_user is not None and hasattr(current_user, 'id'):
+        return redirect("/", code=302)
     form = UserForm()
-    if request.method == 'POST':
-
-        if form.validate_on_submit():
-            new_user = User()
-            form.populate_obj(new_user)
-            new_user.set_password(form.password.data) #pw should be hashed with some salt
-            db.session.add(new_user)
-            db.session.commit()
-            return redirect('/users')
-
+    if request.method == 'POST' and form.validate_on_submit():
+            q = db.session.query(User).filter(User.email == form.email.data)
```

Comment:
Instead of querying if that email is already present we might impose email uniqueness in the database and catch the exception during db.session.commit().
SQLalchemy seems to throw expections if integrity checks fail https://stackoverflow.com/questions/10322514/dealing-with-duplicate-primary-keys-on-insert-in-sqlalchemy-declarative-style.

Anyway, I don't know how hard/annoying this is to actually implement, so up to you to decide.

Reply:
Working on that. Tomorrow i will decide on that.

Reply:
https://stackoverflow.com/questions/5729500/how-does-sqlalchemy-handle-unique-constraint-in-table-definition
I've already tried to insert multiple entries with same email but no exception are thrown.

Reply:
Have you tried putting this in the email field https://docs.sqlalchemy.org/en/13/core/constraints.html#unique-constraint?

Reply:
I've done it. The problem was that the database that is in the repo was already initialized with the email column that was not unique. For this reason the unique constraint was not respected.
I update and push the code.

Reply:
(if with that it still doesn't work let's just push the code as it is)

Reply:
> The problem was that the database that is in the repo was already initialized with the email column that was not unique.

Haha, nice catch!